internal/olm/operator/registry/index_image.go

```diff
@@ -146,8 +478,41 @@ func (c IndexImageCatalogCreator) UpdateCatalog(ctx context.Context, cs *v1alpha
 			// if no annotations exist and image reference is empty, error out
 			return errors.New("cannot upgrade: no catalog image reference exists in catalog source spec or annotations")
 		}
+
 		// if no annotations exist and image reference exists, set it to index image
 		c.IndexImage = cs.Spec.Image
+
+		// check if index image adopts File-Based Catalog or SQLite index image format
+		isFBCImage, err := isFBC(ctx, c.IndexImage)
+		if err != nil {
+			return err
+		}
+		c.HasFBCLabel = isFBCImage
+
+		// Upgrade a bundle that was installed using OLM
+		if c.HasFBCLabel {
+			// Upgrading when installed traditionally by OLM
+			upgradedFBC, err := handleTraditionalUpgrade(ctx, c.IndexImage, c.BundleImage, subscription.Spec.Channel)
+			if err != nil {
+				return err
+			}
+			c.FBCContent = upgradedFBC
+		}
+	} else {
+		// check if index image adopts File-Based Catalog or SQLite index image format
+		isFBCImage, err := isFBC(ctx, c.IndexImage)
+		if err != nil {
+			return err
+		}
+		c.HasFBCLabel = isFBCImage
+
+		// Upgrade an installed bundle from catalog source annotations
+		if c.HasFBCLabel {
+			err = setupFBCupdates(&c, ctx)
+			if err != nil {
+				return err
+			}
+		}
 	}
 
 	c.setAddMode()
```

Comment:
Should we be doing this by default? Or only for SQLite based images?

I think only for the latter.

Reply:
It would not matter, since we changed the container creation command to no longer use `opm registry add`. For the FBC case, we don't run any command that involves the mode flag. If it is specified, it will be ignored.

Another option is to error out early on in the process. This would be erroring out if an FBC index image was used (or default) **AND** the mode flag was specified.

Reply:
We should error out early when we detect FBC index and a user provides a mode to insert the bundle in.